### PR TITLE
js files updated and based off of A24, not A23.

### PR DESCRIPTION
**Pull request by Jammyjamjamman**
All the code files have been rebased + relocated, based on A24 code.

### CODE REVIEW

**Comment by andy5995** [comment]
@Jammyjamjamman Am I correct in assuming you wanted to improve the code for the a24 version before you worked on the a25 version? I'll give it a test later tonight or tomorrow.

@JustusAvramenko would you like to review this before its merged?

**Review by andy5995 — CHANGES_REQUESTED** [review]
@Jammyjamjamman How about bumping the version in mod.json...

**Comment by Jammyjamjamman** [comment]
Bumped the version.

File: a24/simulation/helpers/InitGame.js [hunk 1]
```diff
@@ -1,64 +1,101 @@
...
+	 // Map or player data (handicap...) dependent initialisations of components (i.e. garrisoned units).
+	 Engine.BroadcastMessage(MT_InitGame, {});
+ 
+	 cmpAIManager.TryLoadSharedComponent();
+	 cmpAIManager.RunGamestateInit();
+ }
+ 
+ Engine.RegisterGlobal("PreInitGame", PreInitGame);
+ Engine.RegisterGlobal("InitGame", InitGame);
+
```
Comment (andy5995) [inline a24/simulation/helpers/InitGame.js]: @Jammyjamjamman can you set your editor to make sure there's a newline at the end of the file?

https://unix.stackexchange.com/a/18789

**Review by andy5995 — APPROVED** [review]
I tested this and it works on A24 with no warnings or errors.

The files should probably have newlines at the end

File: a24/simulation/helpers/InitGame.js (hunk 1 above)
Comment (Jammyjamjamman) [reply a24/simulation/helpers/InitGame.js]: done

**Pull request merged**